custom_components/toon_smartmeter/sensor.py

```diff
@@ -150,6 +150,10 @@ def __init__(self, data, name, sensor_type, unit, icon):
         self._discovery = False
         self._dev_id = {}
 
+    def _validateOutput(self, value):
+        """Return 0 if the output from the Toon is NaN (happens after a reboot)"""
+        return (0 if value == "NaN" else value)
+
```

Comment:
@Wesley-Vos Thanks for the PR! I read somewhere some users also have 'nan' as value instead of 'NaN' so maybe it's better to do a case insensitive check.
```python
if value.lower() == "nan":
  return 0

return value
```
And maybe it's better (more performant) to replace all occurrences of NaN/nan in the JSON data string before parsing, but this fine too.

Reply:
Hi! I really like your suggestion to do a replace operation on the JSON string. I will implement it.